Hex Calculator/HexCalc.java

```diff
@@ -0,0 +1,446 @@
+        return (addHexadecimalnumbers);
+    }
+    
+    /**
+     * Subtraction of two Hexadecimal numbers
+     * @param firstHexaNumber denotes first Hexadecimal number
+     * @param secondHexaNumber denotes second Hexadecimal number
+     * @return Subtracted Hexadecimal numbers in the form of a string
+     */
+    @Override
+    public String subtractNumbers(String firstHexaNumber,String secondHexaNumber)
+    {
+        int firstDecimalNumber=Integer.parseInt(this.convertToDecimal(firstHexaNumber)),secondDecimalNumber=Integer.parseInt(this.convertToDecimal(secondHexaNumber)),subtractDecimalNumber=(firstDecimalNumber-secondDecimalNumber);
+        String subtractHexadecimalnumbers=null;
+        if(subtractDecimalNumber<0)
+        {
+            subtractHexadecimalnumbers=Integer.toString(-1*subtractDecimalNumber);
+            subtractHexadecimalnumbers=convertToHexaDecimal(subtractHexadecimalnumbers);
+            return ("-"+subtractHexadecimalnumbers);
+        }
+        else if(subtractDecimalNumber>0)
+        {
+            subtractHexadecimalnumbers=Integer.toString(subtractDecimalNumber);
+            subtractHexadecimalnumbers=convertToHexaDecimal(subtractHexadecimalnumbers);
+            return (subtractHexadecimalnumbers);
+        }
+        else
+            return "0";
+    }
+    
+    /**
+     * Multiplication of two Hexadecimal numbers
+     * @param firstHexaNumber denotes first Hexadecimal number
+     * @param secondHexaNumber denotes second Hexadecimal number
+     * @return Multiplied Hexadecimal numbers in the form of a string
+     */
+    @Override
+    public String multiplyNumbers(String firstHexaNumber,String secondHexaNumber)
+    {
+        int firstDecimalNumber=Integer.parseInt(this.convertToDecimal(firstHexaNumber)),secondDecimalNumber=Integer.parseInt(this.convertToDecimal(secondHexaNumber)),multiplyDecimalNumber=(firstDecimalNumber*secondDecimalNumber);
+        String multiplyHexadecimalnumbers=Integer.toString(multiplyDecimalNumber);
+        multiplyHexadecimalnumbers=convertToHexaDecimal(multiplyHexadecimalnumbers);
+        return (multiplyHexadecimalnumbers);
+    }
+    
+    /**
+     * Division of two Hexadecimal numbers
+     * @param firstHexaNumber denotes first Hexadecimal number
+     * @param secondHexaNumber denotes second Hexadecimal number
+     * @return Divided Hexadecimal numbers in the form of a string
+     */
+    @Override
+    public String divideNumbers(String firstHexaNumber,String secondHexaNumber)
+    {
+        int firstDecimalNumber=Integer.parseInt(this.convertToDecimal(firstHexaNumber)),secondDecimalNumber=Integer.parseInt(this.convertToDecimal(secondHexaNumber)),divideDecimalNumber=(firstDecimalNumber/secondDecimalNumber);
+        String divideHexadecimalnumbers=Integer.toString(divideDecimalNumber);
+        divideHexadecimalnumbers=convertToHexaDecimal(divideHexadecimalnumbers);
+        return (divideHexadecimalnumbers);

```

Comment:
return convertToHexaDecimal(Integer.toString(firstDecimalNumber/secondDecimalNumber));
So we have saved new variable creation and also some lines of code.